tests/conftest.py

```diff
@@ -2391,14 +2389,49 @@ def titanic_pandas_data_context_with_v013_datasource_with_checkpoints_v1_with_em
             my_runtime_data_connector:
                 module_name: great_expectations.datasource.data_connector
                 class_name: RuntimeDataConnector
-                runtime_keys:
+                batch_identifiers:
                     - pipeline_stage_name
                     - airflow_run_id
         """
 
     context.test_yaml_config(
         name="my_datasource", yaml_config=datasource_config, pretty_print=False
     )
+
+    if (
+        any(
+            [
+                dbms in test_backends
+                for dbms in ["postgresql", "sqlite", "mysql", "mssql"]
+            ]
+        )
+        and (sa is not None)
+        and is_library_loadable(library_name="sqlalchemy")
+    ):
+        db_file = file_relative_path(
+            __file__,
+            "test_sets/test_cases_for_sql_data_connector.db",
+        )
+        datasource_config = f"""
+            class_name: Datasource
+
+            execution_engine:
+                class_name: SqlAlchemyExecutionEngine
+                connection_string: sqlite:///{db_file}
```

Comment:
Is it intended that we check for any of postgres, sqlite, mysql, and mssql on 2402, but then only create a datasource for sqlite here?

Reply:
Unsure. @alexsherstinsky ^^

Reply:
@roblim The line numbers have changed; if this is about the fixture `data_context_with_sql_datasource_for_testing_get_batch()` then the idea was that we prepared an `SQLite` file containing all the needed tables (because `SQLite` makes this so easy to do!).  We commit this file into the repository and use it as a fixture.  If this fixture needs to change, we want to be careful about doing so, since different tests depend on it.  Another example of taking advantage of this capability of `SQLite` appears in `tests/cli/test_checkpoint.py` (please see the `titanic_data_context_with_sql_datasource()` fixture).  Thanks.

Reply:
To confirm, should we then be checking for databases that are not sqlite? Is there ever a situation where we might have postgres but not sqlite in a `test_backends`? If so, could there be issues in this case?